TeamCode/DistanceSensorAutonomous.java

```diff
@@ -0,0 +1,211 @@
+        target = (int) Math.ceil(Math.random() * 3);
+        telemetry.addData("target", target);
+        telemetry.update();
+
+        if (target == 1) {
+            targetXMin = 0;
+            targetYMax = 0;
+            targetXMax = 15;
+            targetYMax = 15;
+        } else if (target == 2) {
+            targetXMin = 60;
+            targetYMin = 60;
+            targetXMax = 75;
+            targetYMax = 75;
+        } else {
+            targetXMin = 0;
+            targetYMin = 135;
+            targetXMax = 15;
+            targetYMax = 150;
+        }
+
+        //mecanumDrive.setEndPowerBehaviorBrake();
+        waitForStart();
+        while(opModeIsActive()){
+                //Sets Robot on the line after it looks at the rings
+                while(rightDistance.getDistance(DistanceUnit.CM)  <= 95.5){
+                    driveLeft(0.4);
+                }
+                while(rightDistance.getDistance(DistanceUnit.CM) >= 95.5){
+                    driveRight(0.2);
+                }
+
+                //Robot moves to Area and posts "Dropped the thing"
+                if (target == 3) {
+                    while (colorSensor.blue() < 200 &&
+                            colorSensor.green() < 200 &&
+                            colorSensor.red() < 200){
+                        driveForward(0.3);
+                    }
+                }
+                else{
+                    while (frontDistance.getDistance(DistanceUnit.CM) > targetYMax) {
+                        driveForward(1);
+                    }
+                    while (frontDistance.getDistance(DistanceUnit.CM) < targetYMin) {
+                        driveBackwards(1);
+                    }
+                }
+                while (rightDistance.getDistance(DistanceUnit.CM) > targetXMax) {
+                    driveRight(1);
+                }
+                while (rightDistance.getDistance(DistanceUnit.CM) < targetXMin) {
+                    driveLeft(1);
+                }
+                fullStop();
+                telemetry.addData("Dropped the thing", "");
+                telemetry.update();
+                sleep(200);
+
+                if (target == 3){
+                    break;
+                }
+
+                //Arm Moves
+                armSetOut(0.5);
+                //Position 0 is Retract
+                handServo.setPosition(0);
+                sleep(100);
+                driveLeft(1);
+                sleep(100);
+                returnArm();
+
+                //Find actual color sensor values
+                while (colorSensor.blue() < 300 &&
+                        colorSensor.green() < 300 &&
+                        colorSensor.red() < 300) {
+                    driveBackwards(.1);
+                }
+
+                fullStop();
+                telemetry.addData("finished", "");
+                telemetry.update();
+                break;
+            }
+        }
+
+    private void Initiate(){
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Reply:
I’m sorry. Can you be more specific. Are you talking about the Color sensor object using red(). If so, we compete in a competition called First Tech Robotics and that method comes from their API which we have no control over

Reply:
I appreciate you taking the time to help a new programmer learn the standard conventions for Java

Reply:
Happy to help.